src/static/register.php

```diff
@@ -2,97 +2,88 @@
-            }
 
-            // Close statement
-            mysqli_stmt_close($stmt);
-        }
+    //check for empty email
+    if(empty(trim($_POST["useremail"]))){
+        $useremail_err = "Please enter a valid Email address.";
+    }  
+    //check for email format
+    if (!filter_var($useremail, FILTER_VALIDATE_EMAIL)) {
+    $useremail_err = "Invalid email format"; 
+    
     }
     
-    // Validate password
+        // Validate password
     if(empty(trim($_POST["password"]))){
         $password_err = "Please enter a password.";     
     } elseif(strlen(trim($_POST["password"])) < 6){
         $password_err = "Password must have atleast 6 characters.";
     } else{
         $password = trim($_POST["password"]);
-    }
+        }
     
-    // Validate confirm password
+        // Validate confirm password
     if(empty(trim($_POST["confirm_password"]))){
         $confirm_password_err = "Please confirm password.";     
     } else{
         $confirm_password = trim($_POST["confirm_password"]);
         if(empty($password_err) && ($password != $confirm_password)){
             $confirm_password_err = "Password did not match.";
+            }
         }
-    }
     
-    // Check input errors before inserting in database
-    if(empty($username_err) && empty($password_err) && empty($confirm_password_err)){
+
+    //data validation and sql entry
+     if(empty($useremail_err) && empty($password_err) && empty($confirm_password_err)){
+        $sql = "INSERT INTO user_data (useremail, password1) VALUES (?, ?)";
+            if($stmt = mysqli_prepare($link, $sql)){
+                // Bind variables to the prepared statement as parameters
+                mysqli_stmt_bind_param($stmt, "ss", $param_useremail, $param_password);
+
+                $param_useremail = $useremail;
+                $param_password = password_hash($password, PASSWORD_DEFAULT); // Creates a password hash
         
-        // Prepare an insert statement
-        $sql = "INSERT INTO users (username, password) VALUES (?, ?)";
-         
-        if($stmt = mysqli_prepare($link, $sql)){
-            // Bind variables to the prepared statement as parameters
-            mysqli_stmt_bind_param($stmt, "ss", $param_username, $param_password);
-            
-            // Set parameters
-            $param_username = $username;
-            $param_password = password_hash($password, PASSWORD_DEFAULT); // Creates a password hash
+                
             
-            // Attempt to execute the prepared statement
-            if(mysqli_stmt_execute($stmt)){
-                // Redirect to login page
-                header("location: login.php");
-            } else{
-                echo "Something went wrong. Please try again later.";
+                // Attempt to execute the prepared statement
+                if(mysqli_stmt_execute($stmt)){
+                    // Redirect to login page
```

Comment:
im not sure if we should redirect them to the login page or just log them in automatically here and send them to the home page.